native2ika/fetcher.go

```diff
@@ -0,0 +1,20 @@
+package native2ika
+
+//go:generate msgp
+
+// SignReq represents a signature request.
+type SignReq struct {
+	ID        uint64 `msg:"id"`
+	Payload   []byte `msg:"payload"`
+	DWalletID string `msg:"dwallet_id"`
+	UserSig   string `msg:"user_sig"`
+	FinalSig  []byte `msg:"final_sig"`
```

Comment:
What is final sign?

Reply:
final sig is what IKA returns after signing the message

Reply:
you can add comments to the fields:
```
// Payload is bytes that have to be signed
Payload []byte
// UserSig is the payload signature using with user private key share
UserSig []byte
```

Reply:
BTW, I think we don't need the tags ``msg:"final_sig"` - it should work out of the box.

Reply:
Yes, this is not a part of the sign request.

Reply:
Removed it

Reply:
Also added the comments